backend/src/test/java/com/example/backend/service/TripServiceTest.java

```diff
@@ -39,14 +47,79 @@ void getAll_returnTripsFromRepo() {
     }
 
     @Test
-    void add_addsTripToRepoAndReturnsTrip() {
-        // given
+    void add_automaticallyInsertsTimeStampWhenTripAdded_WithoutLocationIDs() {
+        //Given
+        Trip trip = new Trip("abc1", "My Trip", List.of(
+                new Location("Kölner Dom", 50.941386546092225, 6.958270670147375),
+                new Location("Planten un Blomen", 53.5625456617408, 9.98188182570993)
+        ));
+
+        Trip tripWithTestTimeStamp = new Trip("abc1", testTimeStamp(), "My Trip", List.of(
+                new Location("Kölner Dom", 50.941386546092225, 6.958270670147375),
+                new Location("Planten un Blomen", 53.5625456617408, 9.98188182570993)
+        ));
+
+        TripRepo tripRepo = mock(TripRepo.class);
+        when(tripRepo.save(trip)).thenReturn(trip);
+
+        IdGenerator idGenerator = mock(IdGenerator.class);
+
+        TimeStampGenerator timeStampGenerator = mock(TimeStampGenerator.class);
+        when(timeStampGenerator.generateTimeStamp())
+                .thenReturn(LocalDateTime.of(2020, 1, 1, 0, 0));
```

Comment:
Das is hier quasi das selbe wie der timestamp, den du in deinem tripWithTestTimeStamp hast oder? 

Was hälst du davon? 
```java
when(timeStampGenerator.generateTimeStamp()).thenReturn(tripWithTimestamp.getTripTimestamp());
```
?